include/oneapi/dpl/pstl/parallel_backend_utils.h

```diff
@@ -279,57 +286,66 @@ __set_intersection_construct(_ForwardIterator1 __first1, _ForwardIterator1 __las
 }
 
 template <typename _ForwardIterator1, typename _ForwardIterator2, typename _OutputIterator, typename _Compare,
-          typename _CopyConstructRange>
+          typename _CopyConstructRange, typename _Proj1 = oneapi::dpl::identity,
+          typename _Proj2 = oneapi::dpl::identity>
 _OutputIterator
 __set_difference_construct(_ForwardIterator1 __first1, _ForwardIterator1 __last1, _ForwardIterator2 __first2,
                            _ForwardIterator2 __last2, _OutputIterator __result, _Compare __comp,
-                           _CopyConstructRange __cc_range)
+                           _CopyConstructRange __cc_range, _Proj1 __proj1 = {}, _Proj2 __proj2 = {})
 {
     using _Tp = typename ::std::iterator_traits<_OutputIterator>::value_type;
 
+    auto __proj1_deref = [__proj1](_ForwardIterator1 __it) { return std::invoke(__proj1, *__it); };
+    auto __proj2_deref = [__proj2](_ForwardIterator2 __it) { return std::invoke(__proj2, *__it); };
```

Comment:
more places for the projected deref callable struct here and later in this file.

Reply:
This is used only in the host backends, so using structs is not necessary.

Reply:
Please see my comment at https://github.com/uxlfoundation/oneDPL/pull/2320#discussion_r2303228180

Reply:
Fixed.